src/base_object.h

```diff
@@ -65,6 +65,11 @@ class BaseObject {
   static inline void WeakCallback(
       const v8::WeakCallbackInfo<Type>& data);
 
+  // persistent_handle_ needs to be at a fixed offset from the start of the
+  // class because it is used by tools/gen-postmortem-metadata.py to calculate
+  // offsets and generate debug symbols for BaseObject, which assumes that the
+  // position of members in memory are predictable. Because of that, there must
+  // be no variable-size members before persistent_handle_
```

Comment:
These comments are subtly wrong because there is no such thing as a variable-sized member.  What I suspect you're getting at is data members that are conditionally compiled in based on preprocessor defines.

Reply:
Variable-sized members can happen when a member type is based on the class template (which is not the case for BaseObject right now, but the comment is here to avoid problems if this changes in the future).

But I agree, this explanation should be improved. Maybe this could be explained in `doc/guides/node-postmortem-support.md` to avoid repetitive comments and to explain it better? Then this comment could be:

```c++
  // persistent_handle_ needs to be at a fixed offset from the start of the
  // class because it is used by tools/gen-postmortem-metadata.py to calculate
  // offsets and generate postmortem metadata for BaseObject. For more
  // information please refer to `doc/guides/node-postmortem-support.md`
```

What do you think?

Reply:
> Variable-sized members can happen when a member type is based on the class template

Okay, I understand what you mean now but I wouldn't use the word "variable-sized" for that, that's too easy to confuse with variable-length arrays and structs, but they are quite different beasts.

(And also because it's still subtly wrong: there's nothing variable-sized about them, `BaseObject<Foo>` and `BaseObject<Bar>` are simply different types.)

Reply:
@bnoordhuis Maybe we can just leave it as `persistent_handle_ needs to be at a fixed offset from the start of the class.....which assumes the position of members in memory are predictable`, I think that is already clear enough.